pkg/agent/openflow/pipeline.go

```diff
@@ -2208,53 +2162,58 @@ func (c *client) genPacketInMeter(meterID binding.MeterIDType, rate uint32) bind
+		updateBindingTable(c.bridge, sessionAffinityTable, binding.LastTableID, binding.TableMissActionNone)
+		updateBindingTable(c.bridge, serviceLBTable, endpointDNATTable.GetID(), binding.TableMissActionNext)
+		updateBindingTable(c.bridge, endpointDNATTable, c.egressEntryTable, binding.TableMissActionNext)
+		updateBindingTable(c.bridge, conntrackCommitTable, hairpinSNATTable.GetID(), binding.TableMissActionNext)
+		updateBindingTable(c.bridge, hairpinSNATTable, L2ForwardingOutTable.GetID(), binding.TableMissActionNext)
 	} else {
-		c.pipeline[spoofGuardTable] = bridge.CreateTable(spoofGuardTable, conntrackTable, binding.TableMissActionDrop)
-		c.pipeline[ipv6Table] = bridge.CreateTable(ipv6Table, conntrackTable, binding.TableMissActionNext)
-		c.pipeline[conntrackStateTable] = bridge.CreateTable(conntrackStateTable, dnatTable, binding.TableMissActionNext)
-		c.pipeline[dnatTable] = bridge.CreateTable(dnatTable, c.egressEntryTable, binding.TableMissActionNext)
-		c.pipeline[conntrackCommitTable] = bridge.CreateTable(conntrackCommitTable, L2ForwardingOutTable, binding.TableMissActionNext)
+		updateBindingTable(c.bridge, spoofGuardTable, conntrackTable.GetID(), binding.TableMissActionDrop)
+		updateBindingTable(c.bridge, ipv6Table, conntrackTable.GetID(), binding.TableMissActionNext)
+		updateBindingTable(c.bridge, conntrackStateTable, dnatTable.GetID(), binding.TableMissActionNext)
+		updateBindingTable(c.bridge, dnatTable, c.egressEntryTable, binding.TableMissActionNext)
+		updateBindingTable(c.bridge, conntrackCommitTable, L2ForwardingOutTable.GetID(), binding.TableMissActionNext)
 	}
 	// The default SNAT is implemented with OVS on Windows.
 	if c.enableEgress || runtime.IsWindowsPlatform() {
-		c.pipeline[snatTable] = bridge.CreateTable(snatTable, l2ForwardingCalcTable, binding.TableMissActionNext)
+		updateBindingTable(c.bridge, snatTable, l2ForwardingCalcTable.GetID(), binding.TableMissActionNext)
 	}
 	if runtime.IsWindowsPlatform() {
-		c.pipeline[uplinkTable] = bridge.CreateTable(uplinkTable, spoofGuardTable, binding.TableMissActionNone)
+		updateBindingTable(c.bridge, uplinkTable, spoofGuardTable.GetID(), binding.TableMissActionNone)
 	}
 	if c.enableAntreaPolicy {
-		c.pipeline[AntreaPolicyEgressRuleTable] = bridge.CreateTable(AntreaPolicyEgressRuleTable, EgressRuleTable, binding.TableMissActionNext)
-		c.pipeline[AntreaPolicyIngressRuleTable] = bridge.CreateTable(AntreaPolicyIngressRuleTable, IngressRuleTable, binding.TableMissActionNext)
+		updateBindingTable(c.bridge, AntreaPolicyEgressRuleTable, EgressRuleTable.GetID(), binding.TableMissActionNext)
+		updateBindingTable(c.bridge, AntreaPolicyIngressRuleTable, IngressRuleTable.GetID(), binding.TableMissActionNext)
 	}
 }
 
+func updateBindingTable(bridge binding.Bridge, table binding.Table, nextID uint8, missAction binding.MissActionType) binding.Table {
+	table.SetNext(nextID)
```

Comment:
What you think a single func that takes all the parameters and create the table on bridge?

Reply:
I think it is more flexible if one function does only one thing. For example, we might expect to change the next table in the pipeline after new feature is introduced, then we could only call SetNext, and don't need to reset all of nextTable, missAction, and recreate table on the bridge.

Reply:
How about this one?

Reply:
I agree, but the question for the Create() func should it handle all necessary operations together (meanwhile we can always have separate funcs for other uses)?

Reply:
OK, I will try to add a new function in binding.Table to involve all these property settings.